IntegrationTests/UI.IntegrationTests/CommandsDialogs/FormFileHistoryTests.cs

```diff
@@ -39,6 +39,47 @@ public void File_history_should_behave_as_expected(string fileRelativePath, stri
             }, Path.Combine(fileRelativePath, fileName), commands);
     }
 
+    [Test]
+    public void File_history_should_display_contents_in_all_revisions()
+    {
+        const string text = "This text will appear in a file that shall be renamed";
+
+        using ReferenceRepository referenceRepository = new(createCommit: false);
+        Console.WriteLine(referenceRepository.Module.WorkingDir);
+        GitUICommands commands = new(GlobalServiceContainer.CreateDefaultMockServiceContainer(), referenceRepository.Module);
+
+        string initialRevision = referenceRepository.CreateCommitRelative(string.Empty, "a.txt", "Create 'a.txt'", text);
+        string renameRevision = referenceRepository.RenameRepoFile(string.Empty, "a.txt", "b.txt", commitMessage: "Rename to 'b.txt'");
+        string anotherRenameRevision = referenceRepository.RenameRepoFile(string.Empty, "b.txt", "c.txt", commitMessage: "Rename to 'c.txt'");
+
+        AppSettings.FollowRenamesInFileHistory = true;
+
+        RunFormTest(
+            form =>
+            {
+                bool textLoaded;
+
+                WaitForRevisionsToBeLoaded(form);
+                form.GetTestAccessor().SelectViewTab();
+                form.GetTestAccessor().FileViewer.TextLoaded += (_, _) => textLoaded = true;
+
+                textLoaded = false;
+                form.GetTestAccessor().RevisionGrid.SetSelectedRevision(ObjectId.Parse(initialRevision)).Should().BeTrue();
+                UITest.ProcessUntil("Loading file text", () => textLoaded);
+                form.GetTestAccessor().FileViewer.GetTestAccessor().FileViewerInternal.GetTestAccessor().TextEditor.Text.Should().Be(text);
```

Comment:
💭  This is repeated several times, may worth considering refactoring this block as a local function.